src/language_server_protocol.rs

```diff
@@ -1824,6 +1824,32 @@ impl LanguageClient {
         Ok(())
     }
 
+    pub fn window_showMessageRequest(&self, params: &Value) -> Fallible<Value> {
+        info!("Begin {}", lsp::request::ShowMessageRequest::METHOD);
+        let msg_params: ShowMessageRequestParams = params.clone().to_lsp()?;
+        let msg_actions = msg_params.actions.unwrap_or_default();
+        let n_actions = msg_actions.len();
+        let actions: Vec<Value> = msg_actions
+            .into_iter()
+            .enumerate()
+            .map(|(i, item)| Value::String(format!("{}) {}", i + 1, item.title)))
+            .collect();
+        let mut options: Vec<Value> = Vec::with_capacity(actions.len() + 1);
+        options.push(Value::String(msg_params.message));
+        options.extend(actions);
+
+        let mut v = Value::Null;
+        let result: Option<i64> = self.vim()?.rpcclient.call("s:inputlist", options)?;
+        if let Some(answer) = result {
+            if answer > 0 && answer < (n_actions - 1) as i64 {
```

Comment:
The second portion should be something like `<= n_actions`, right?

Say there are 2 actions, `2` should be a legitimate answer.

Reply:
The mapping between actions and parameters to ``inputlist`` isn't 1 to 1, they're actually shifted to accommodate the prompt. ``params.message`` is actually the index 0, so when the user types 1, they're actually selecting 0.

See screenshot:

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/108725/53933993-d4a9f880-406e-11e9-90f4-2d99d90c3838.png">

We're instructing the user to type from 1 to 3, but the user could actually type 0 and get an invalid option. Vim also supports mouse click, so that's why we can't simply say that the second option is index 0.

Reply:
Yes, I get that part.

What I meant is, say in this screenshot, if I select or click 3, it should still be an acceptable answer, right? But in your code, `3 < (3 - 1)`, which is returning false.

Reply:
Oh shoot you're right I got this wrong, my bad.